src/ai/backend/gateway/logs.py

```diff
@@ -0,0 +1,203 @@
+            query = query.where(where)
+            count_query = query.where(where)
+
+        result = await conn.execute(query)
+        async for row in result:
+            result_item = {
+                'log_id': str(row['id']),
+                'created_at': datetime.timestamp(row['created_at']),
+                'severity': row['severity'],
+                'source': row['source'],
+                'user': row['user'],
+                'is_read': row['is_read'],
+                'message': row['message'],
+                'context_lang': row['context_lang'],
+                'context_env': row['context_env'],
+                'request_url': row['request_url'],
+                'request_status': row['request_status'],
+                'traceback': row['traceback'],
+            }
+            if result_item['user'] is not None:
+                result_item['user'] = str(result_item['user'])
+            if is_admin:
+                result_item['is_cleared'] = row['is_cleared']
+            resp['logs'].append(result_item)
+        resp['count'] = await conn.scalar(count_query)
+        if params['mark_read'].lower() == 'true':
+            update = (sa.update(error_logs)
+                        .values(is_read=True)
+                        .where(error_logs.c.id.in_([x['log_id'] for x in resp['logs']])))
+            await conn.execute(update)
+        return web.json_response(resp, status=200)
+
+
+@auth_required
+@server_status_required(READ_ALLOWED)
+async def clear(request: web.Request) -> web.Response:
+    dbpool = request.app['dbpool']
+    domain_name = request['user']['domain_name']
+    user_role = request['user']['role']
+    user_uuid = request['user']['uuid']
+    log_id = uuid.UUID(request.match_info['log_id'])
+
+    log.info('CLEAR')
+    async with dbpool.acquire() as conn, conn.begin():
+        query = (sa.update(error_logs)
+                   .values(is_cleared=True))
+        if request['is_superadmin']:
+            query = query.where(error_logs.c.id == log_id)
+        elif user_role == UserRole.ADMIN or user_role == 'admin':
+            j = (groups.join(agus, groups.c.id == agus.c.group_id))
+            usr_query = (sa.select([agus.c.user_id])
+                           .select_from(j)
+                           .where([groups.c.domain_name == domain_name]))
+            result = await conn.execute(usr_query)
+            usrs = await result.fetchall()
+            user_ids = [g.id for g in usrs]
+            query = query.where((error_logs.c.user_id.in_(user_ids)) &
+                                (error_logs.c.id == log_id))
+        else:
+            query = (query.where((error_logs.c.user_id == user_uuid) &
+                                 (error_logs.c.id == log_id)))
+
+        result = await conn.execute(query)
+        assert result.rowcount == 1
+
+        return web.json_response({'success': True}, status=200)
+
+
+async def init(app: web.Application) -> None:
+    pass
```

Comment:
We should add a periodic timer task which cleans up old-enought logs (e.g., appended before 3 months).